packages/agent-toolkit/src/ai-sdk/toolkit.ts

```diff
@@ -0,0 +1,89 @@
+import type {
+  Tool as CoreTool,
+  LanguageModelV1Middleware,
+  LanguageModelV1StreamPart,
+} from "ai";
+
+import RecallAPI from "../shared/api.js";
+import { type Configuration, isToolAllowed } from "../shared/configuration.js";
+import { tools } from "../shared/tools.js";
+import RecallTool from "./tool.js";
+
+/**
+ * A toolkit for the Recall agent.
+ * @example
+ * ```ts
+ * const toolkit = new RecallAgentToolkit({
+ *   privateKey: "0x...",
+ *   configuration: {
+ *     actions: {
+ *       account: {
+ *         read: true,
+ *         write: true,
+ *       },
+ *     },
+ *   },
+ * });
+ * ```
+ */
+export default class RecallAgentToolkit {
+  private _recall: RecallAPI;
+
+  tools: { [key: string]: CoreTool };
+
+  constructor({
+    privateKey,
+    configuration,
+  }: {
+    privateKey: string;
+    configuration: Configuration;
+  }) {
+    this._recall = new RecallAPI(privateKey, configuration.context);
+    this.tools = {};
+
+    const filteredTools = tools.filter((tool) =>
+      isToolAllowed(tool, configuration),
+    );
+
+    filteredTools.forEach((tool) => {
+      this.tools[tool.method] = RecallTool(
+        this._recall,
+        tool.method,
+        tool.description,
+        tool.parameters,
+      );
+    });
+  }
+
+  middleware(): LanguageModelV1Middleware {
```

Comment:
can you explain why we need this? just because of the framework agnostic factor? because in this case, we're not _actually_ providing any "middleware" functionality, we're just passing things right along...